code/modules/projectiles/guns/projectile/automatic.dm

```diff
@@ -1,189 +1,222 @@
+	..()
+	icon_state = "[initial(icon_state)][stored_magazine ? "-[stored_magazine.max_ammo]" : ""][chambered ? "" : "-e"]"
+	return
+
+/obj/item/weapon/gun/projectile/automatic/Fire(atom/target as mob|obj|turf|area, mob/living/user as mob|obj, params, reflex = 0, struggle = 0)
+	if(burst_fire)
+		if(!ready_to_fire() || (!burst_ready_to_fire() && !burst_busy))
+			return
+		fire_delay = burst_firerate - 1 //Sanity needed
+		burst_busy = 1
+		var/shots_fired = 0 //Haha, I'm so clever
+		var/to_shoot = min(burst_count, getAmmo())
+		for(var/i = 1; i <= to_shoot; i++)
+			..()
+			shots_fired++
+			sleep(burst_firerate) //Make sure we don't hit our own cooldown, that would be dumb
+		burst_last_fired = world.time
+		burst_busy = 0
+		if(shots_fired) //Only warn the admins if any shots were fired
+			message_admins("[usr] just shot [shots_fired] bullets in burst fire out of [getAmmo() + shots_fired] from their [src.name].")
+	else
+		fire_delay = initial(fire_delay) //Reset the fire delay
+		..() //Default behavior, single shot, let's go
+
+/obj/item/weapon/gun/projectile/automatic/micro_uzi
+	name = "Micro-Uzi"
+	desc = "The most popular pocket-sized submachine gun on the black market. Small magazines for a small weapon, fires quickly. Uses .45 rounds."
+	icon_state = "micro-uzi"
+	inhand_states = list("left_hand" = 'icons/mob/in-hand/left/guninhands_left.dmi', "right_hand" = 'icons/mob/in-hand/right/guninhands_right.dmi')
+	w_class = 2.0
+	max_shells = 12
+	fire_delay = 1
+	burst_count = 4
+	caliber = list(".45" = 1)
+	origin_tech = "combat=5;materials=2;syndicate=8"
+	ammo_type = "/obj/item/ammo_casing/c45"
+	mag_type = "/obj/item/ammo_storage/magazine/uzimicro45"
+
+/obj/item/weapon/gun/projectile/automatic/micro_uzi/isHandgun()
+	return 1
+
+//Based off the FN F2000
+/obj/item/weapon/gun/projectile/automatic/c20r
+	name = "\improper C-20R"
+	desc = "A lightweight burst-fire assault rifle. A favourite of many mercenary operatives, the most famous users of this gun are the mysterious Syndicate. Uses 5.56x45 rounds. Has a 'Scarborough Arms - Per falcis, per pravitas' buttstamp."
+	icon_state = "c20r"
+	inhand_states = list("left_hand" = 'icons/mob/in-hand/left/guns.dmi', "right_hand" = 'icons/mob/in-hand/right/guns.dmi')
+	item_state = "c20r"
+	w_class = 3.0
+	max_shells = 30
+	burst_count = 3
+	caliber = list("556x45" = 1)
+	origin_tech = "combat=5;materials=2;syndicate=8"
+	ammo_type = "/obj/item/ammo_casing/a556x45"
+	mag_type = "/obj/item/ammo_storage/magazine/a556x45"
+	fire_sound = 'sound/weapons/c20r.ogg'
+	fire_sound_far = 'sound/weapons/c20r_far.ogg'
+
+	gun_flags = AUTOMAGDROP | SILENCECOMP
+
+/obj/item/weapon/gun/projectile/automatic/c20r/update_icon()
+	..()
+	overlays = 0
+	if(stored_magazine)
+		icon_state = "c20r-[round(getAmmo(), 5)]"
+	else
+		icon_state = "c20r"
+	if(silenced)
+		overlays += "[icon_state]-silencer"
+	return
+
+/obj/item/weapon/gun/projectile/automatic/xcom
+	name = "\improper XCOM Assault Rifle"
```

Comment:
Just keep it named "assault rifle". It's bad enough that we have too many obvious references.